environment/src/main/java/com/sequenceiq/environment/environment/flow/creation/handler/freeipa/FreeIpaCreationHandler.java

```diff
@@ -114,12 +115,17 @@ public FreeIpaCreationHandler(
     @Override
     public void accept(Event<EnvironmentDto> environmentDtoEvent) {
         EnvironmentDto environmentDto = environmentDtoEvent.getData();
+        //TODO Why? Necessary informations are available in the environmentDto
         Optional<Environment> environmentOptional = environmentService.findEnvironmentById(environmentDto.getId());
         try {
-            if (environmentOptional.isPresent() && environmentOptional.get().isCreateFreeIpa()) {
+            if (environmentOptional.isPresent()) {
                 Environment environment = environmentOptional.get();
-                if (supportedPlatforms.supportedPlatformForFreeIpa(environment.getCloudPlatform())) {
-                    createFreeIpa(environmentDtoEvent, environmentDto, environment);
+                if (Objects.nonNull(environment.getParentEnvironment())) {
+                    attachParentFreeIpaToChild(environmentDtoEvent, environmentDto);
+                } else if (environmentOptional.get().isCreateFreeIpa()) {
+                    if (supportedPlatforms.supportedPlatformForFreeIpa(environment.getCloudPlatform())) {
```

Comment:
Not a big deal, but the nested `if` block could be merged into the enclosing one using a compound condition `...isCreateFreeIpa() && ...supportedPlatformForFreeIpa(...)`.

Reply:
Thank you. Fixed